sympy/matrices/matrices.py

```diff
@@ -1495,6 +1496,189 @@ def is_diagonalizable(self, reals_only=False, **kwargs):
                 ret = False
         return ret
 
+    def _eval_is_positive_definite(self, method="eigen"):
+        """Algorithm dump for computing positive-definiteness of a
+        matrix.
+
+        Parameters
+        ==========
+
+        method : str, optional
+            Specifies the method for computing positive-definiteness of
+            a matrix.
+
+            If ``'eigen'``, it computes the full eigenvalues and decides
+            if the matrix is positive-definite.
+
+            If ``'CH'``, it attempts computing the Cholesky
+            decomposition to detect the definitiveness.
+
+            If ``'LDL'``, it attempts computing the LDL
+            decomposition to detect the definitiveness.
+        """
+        if self.is_hermitian:
+            if method == 'eigen':
+                eigen = self.eigenvals()
+                args = [x.is_positive for x in eigen.keys()]
+                return fuzzy_and(args)
+
+            elif method == 'CH':
```

Comment:
These alternative methods should be tested (or removed if unneeded)

Reply:
I think that there are some slight difference in performance for computing matrix positive definitiveness, though the matrices are too small

```python
>>> from sympy import *
>>> import cProfile

>>> m = Matrix([[2, -1, -1], [-1, 2, -1], [-1, -1, 2]])
>>> cProfile.run("m._eval_is_positive_definite(method='eigen')")
8458 function calls (8272 primitive calls) in 0.006 seconds
>>> cProfile.run("m._eval_is_positive_definite(method='LDL')")
4608 function calls (4537 primitive calls) in 0.003 seconds
>>> cProfile.run("m._eval_is_positive_definite(method='CH')")
11287 function calls (10834 primitive calls) in 0.008 seconds
```

And the inverse result seems true for symbolic matrices
```python
>>> a = Symbol('a', positive=True)
>>> b = Symbol('b', negative=True)
>>> m = Matrix([[a, 0, 0], [0, a, 0], [0, 0, a]])
>>> cProfile.run("m._eval_is_positive_definite(method='eigen')")
1381 function calls (1369 primitive calls) in 0.001 seconds
>>> cProfile.run("m._eval_is_positive_definite(method='LDL')")
12140 function calls (11424 primitive calls) in 0.007 seconds
>>> cProfile.run("m._eval_is_positive_definite(method='CH')")
14466 function calls (13521 primitive calls) in 0.009 seconds
```

In numeric computation, using Cholesky is usually the best answer, however, for symbolic computation, I anticipate a lot of problems can arise because of complex numbers, or symbolic expressions can bloat, so some routine may succeed and other may go extremely slow or fail by cases.

In the result above, `LDL` was the fastest for numerics and `eigenvals` was fastest for symbolics though. And I see the cholesky can be the slowest unlike theoritical time complexity.

So I see things can be multiplexed for shipping public APIs about this. Though I would have to find some good mediocre-sized testing matrices to investigate the pros and cons.